crates/oxc_linter/src/rules/react/no_unescaped_entities.rs

```diff
@@ -0,0 +1,187 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::{self, Error},
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+use phf::{phf_map, Map};
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-react(no-unescaped-entities): `{1}` can be escaped with {2}")]
+#[diagnostic(severity(warning))]
+struct NoUnescapedEntitiesDiagnostic(#[label] pub Span, pub char, pub String);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoUnescapedEntities;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// This rule prevents characters that you may have meant as JSX escape characters from being accidentally injected as a text node in JSX statements.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// JSX escape characters are used to inject characters into JSX statements that would otherwise be interpreted as code.
+    ///
+    /// ### Example
+    /// Incorrect
+    ///
+    /// ```jsx
+    /// <div> > </div>
+    /// ```
+    ///
+    /// Correct
+    ///
+    /// ```jsx
+    /// <div> &gt; </div>
+    /// ```
+    ///
+    /// ```jsx
+    /// <div> {'>'} </div>
+    /// ```
+    NoUnescapedEntities,
+    suspicious
```

Comment:
Shall we try `correctness` and see what happens in real codebases? It feels like this can catch some errors. We can always move this back to suspicious if things go wrong. What do you think?

Reply:
yeah lets do it!